### PR TITLE
Feature/convert to ts (from inside)

### PR DESCRIPTION
**Pull request by soswow**

### CODE REVIEW

File: src/interceptors/fetchInterceptor.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+const fakeResponse = (
+  response: string | any = {},
+  headers: { [k: string]: string } = {}
+) => {
+  // If content type exist and is different to application/json no parse the response
+  if (
+    headers['content-type'] &&
+        headers['content-type'].indexOf('application/json') == -1
+  ) {
+    console.log(response);
```
Comment (zzarcon) [inline src/interceptors/fetchInterceptor.ts]: can you remove this please 🙏

Comment (soswow) [reply src/interceptors/fetchInterceptor.ts]: I did

File: src/Router/index.ts [hunk 2]
```diff
@@ -28,18 +28,16 @@ export interface RouterConfig {
...
       interceptorConfig: Partial<InterceptorConfig<M>> = interceptorDefaultConfig,
-      routerConfig: RouterConfig = { strategies: [] }
+      public routerConfig: RouterConfig = routerDefaultConfig
     ) {
       this.interceptorConfig = merge(
         {},
         interceptorDefaultConfig,
         interceptorConfig
       );
-      this.routerConfig = merge({}, routerDefaultConfig, routerConfig);
```
Comment (soswow) [inline src/Router/index.ts]: So, this was basically merging two lists together, which I don't think what we want.
So, if user defines `routerConfig` with just fetch, this line will merge two arrays and result will be `fetch` + `xhr`

File: src/interceptors/index.ts [hunk 3]
```diff
@@ -2,10 +2,13 @@ import * as xhrInterceptor from './xhrInterceptor';
 import * as fetchInterceptor from './fetchInterceptor';
 
 export interface Interceptors {
-    XMLHttpRequest: typeof xhrInterceptor;
-    fetch: typeof fetchInterceptor;
+  XMLHttpRequest: typeof xhrInterceptor;
+  fetch: typeof fetchInterceptor;
+  http?: any;
+  https?: any;
```
Comment (soswow) [inline src/interceptors/index.ts]: This is needed, because we allows these values to be in the list of `strategies`

**Pull request merged**